pkg/module_manager/models/hooks/global_hook.go

```diff
@@ -0,0 +1,122 @@
+package hooks
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	types2 "github.com/flant/addon-operator/pkg/hook/types"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/models/hooks/kind"
+	"github.com/flant/shell-operator/pkg/hook/types"
+)
+
+// GlobalHook is a representation of the hook, which not belongs to any module
+type GlobalHook struct {
+	executableHook
+	config *GlobalHookConfig
+
+	dc *HookExecutionDependencyContainer
+}
+
+// NewGlobalHook constructs a new global hook
+//
+//	ex - is an executable hook instance (GoHook or ShellHook)
+func NewGlobalHook(ex executableHook, dc *HookExecutionDependencyContainer) *GlobalHook {
+	return &GlobalHook{
+		executableHook: ex,
+		config:         &GlobalHookConfig{},
+
+		dc: dc,
+	}
+}
+
+// InitializeHookConfig initializes the global hook config
+// for GoHook config is precompiled, so we just have to fetch it
+// for ShellHook, that hook will be run with `--config` flag, returns and parses the config
+func (h *GlobalHook) InitializeHookConfig() (err error) {
+	switch hk := h.executableHook.(type) {
+	case *kind.GoHook:
+		cfg := hk.GetConfig()
+		err := h.config.LoadAndValidateGoConfig(cfg)
+		if err != nil {
+			return err
+		}
+
+	case *kind.ShellHook:
```

Comment:
Well, `--config` discovery for go and shell hooks a bit different. So we have to cast it here. Because GoHook has precompiled, during the build-time, config, meanwhile ShellHook has to execute the hook file with `--config` to determine its config

Reply:
No, the comment was about the naming, that this is not required for the hook to be a shell script, it may be just an executable.

Reply:
Oh, got it.
Yup, I only left it compatible with shell-operator and all it's primitives. Because `ExecutableHook` makes more confusion here

Reply:
Also we have stable naming conversion like `shell-lib`, `shell-hook` and so on. I don't see any reason to break it here